### PR TITLE
Rename functions to read CSV

### PR DESCRIPTION
**Pull request by etiennebacher**
Mentioned in https://github.com/pola-rs/r-polars/issues/267#issuecomment-1611369883

This PR:

* `lazy_csv_reader` -> `scan_csv`
* `csv_reader` -> `read_csv`

For now, they only can be accessed with `pl$` because I removed `read_csv_`, but if we still want to have this kind of high-level helper, we should rename it (maybe `read_csv_pl()`?)

Feel free to push directly here

### CODE REVIEW

**Comment by eitsupi** [comment]
Thank you for working on this!

I thought we should do this too, but when I looked at the py-polars implementation I realized that Python's `polars.scan_csv` is completely different from the current R's `lazy_csv_reader`.

That is, Python's `scan_csv` seems to map directly to Rust's `scan_csv`.
https://github.com/pola-rs/polars/blob/d40e403c81620f659244af20ce36470af8afe419/polars/polars-lazy/polars-plan/src/logical_plan/builder.rs#L206

**Comment by sorhawell** [comment]
[actually py-polars uses also LazyCsvReader](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/src/lazyframe.rs#L153) via [here](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/polars/lazyframe/frame.py#L302) and via [here](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/polars/io/csv/functions.py#L680)

I have no idea what the difference between `scan_csv` and `LazyCsvReader`

File: R/info.R [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #' Report information of the package
 #'
 #' @return A list with information of the package
+#' @name polars_info
 #' @examples
 #' polars_info()
-#' @export
-polars_info = function() {
+pl$polars_info = function() {
```
Comment (eitsupi) [inline R/info.R:7]: Could you also update the examples?

**Review by eitsupi — APPROVED** [review]
Thank you for the updates! I think it is a good change for consistency.

Perhaps we can merge this since it seems to have passed the test?

**Comment by eitsupi** [comment]
> [actually py-polars uses also LazyCsvReader](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/src/lazyframe.rs#L153) via [here](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/polars/lazyframe/frame.py#L302) and via [here](https://github.com/pola-rs/polars/blob/1d06aa7f2f1fa7dc4d01f4aa403bc70653c42a3b/py-polars/polars/io/csv/functions.py#L680)

Oh, sorry. I must have misread the code.......

**Review by sorhawell — APPROVED** [review]

**Pull request merged**